namespaces/resource-namespace.md

```diff
@@ -0,0 +1,231 @@
+        {
+            "element": "data-structure"
+        }
+    ]
+}
+```
+
+### Transition (Element)
+
+A transition is an available progression from one state to another state. Exercising a transition initiates a transaction.
+
+The content of this element is array of protocol-specific transactions. At the moment only HTTP protocol is supported.
+
+#### Properties
+
+- `element`: transition (string, fixed)
+- `attributes` (object, required)
+    - `href` (Href, required) - The URI template for this transition.
+    - `relation` - (string) - The relation of this transition from the current resource.  
+    - `parameters` (Href Variables) - Input parameters for this transition.
+    - `attributes` (Data Structure) - Input attributes for this transition.
+- `content` (array[HTTP Transaction]) - Array of transaction examples.
+
+    Transaction examples are protocol-specific examples of REST transaction
+    that was initialized by exercising a transition.
+
+    For the time being this namespace defines only HTTP-specific transaction
+    data structures.
+
+#### Example
+
+```json
+{
+    "element": "transition",
+    "attributes": {
+        "relation": "update",
+        "href": "https://polls.apiblueprint.org/questions/{question_id}"
+    },
+    "content": null
+}
+```
+
+## Protocol-specific Elements
+
+### HTTP Transaction (Element)
+
+Example of an HTTP Transaction. The example's content consist of a reuqest-response
+message pair. A transaction example MUST contain exactly one HTTP request and one HTTP response message.
+
+#### Properties
+
+- `element`: `http-transaction` (string, fixed)
+- `content` (array, fixed) - Request response message pair (tuple).
+
+    The array MUST contain exactly two elements.
+
+    - Elements
+        - (HTTP Request Message)
+        - (HTTP Response Message)
+
+#### Example
+
+```json
+{
+    "element": "http-transaction",
+    "content": [
+        {
+            "element": "http-request",
+            "attributes": {
+                "method": "GET",
+                "href": "https://polls.apiblueprint.org/questions/1"
+            },
+            "content": null
+        },
+        {
+            "element": "http-response",
+            "attributes": {
+                "status": 200,
+                "assets": [
+                    {
+                        "element": "asset",
+                        "meta": {
+                            "class": "message-body"
+                        },
+                        "attributes": {},
+                        "content": "{\"name\": \"John\"}"
+                    }
+                ]
+            },
+            "content": null
+        }
+    ]
+}
+```
+
+### HTTP Message (Element)
+
+- `element`: `http-message` (string, fixed)
+- `attributes`
+    - `headers` (HTTP Headers) - Array of HTTP headers.
+    - `assets` (array[Asset]) - Array of assets associated with the message.
+- `content` (Data Structure)
+
+### HTTP Headers (Object Type)
```

Comment:
I disagree that HTTP headers should be an object type, it should be an **ordered** array of header elements since headers are not uniqued per key.

I'm not certain on the order, but I think we should respect the order given this comment in [RFC2616](http://tools.ietf.org/html/rfc2616)

> The order in which header fields with differing field names are received is not significant. However, it is "good practice" to send general-header fields first, followed by request-header or response- header fields, and ending with the entity-header fields.